src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -181,6 +182,87 @@ public UserDefinedFunction WithBinding(INameResolver nameResolver, IBinderGlue b
             return func;
         }
 
+        private static readonly ISet<string> _restrictedUDFNames = new HashSet<string> { "Type", "IsType", "AsType" };
+
+        // Helper method to create IR UserDefinedFunctions 
+        internal static IEnumerable<UserDefinedFunction> CreateFunctions(IEnumerable<UDF> uDFs, out List<TexlError> errors)
```

Comment:
Moving private method CreateUserDefinedFunctions here. With this hosts can create UDF from parser udf result without being tightly coupled with NamedFormulas

Reply:
This is still private :)
It's public method, but on an internal class.